### PR TITLE
feat: add method for retrieving RichMembers for VO without user attributes

### PR DESCRIPTION
**Pull request by xflord**

### CODE REVIEW

**Review by HejdaJakub — CHANGES_REQUESTED** [review]
I don't think we want to change all these existing methods, especially to implementation with the new required parameter. I thought that we just want one or a few new method(s) (more than one only if we want to allow the definition of attrNames, Vo statuses, etc. - but I am not sure if we want it) with no impact on existing methods. 

According to the task definition, we want this use case only for Vos, so I think you don't have to deal with groups, too.

File: perun-openapi/openapi.yml [hunk 1]
```diff
@@ -9101,6 +9101,22 @@ paths:
         default:
           $ref: '#/components/responses/ExceptionResponse'
 
+  /json/membersManager/getRichMembersWithoutUserAttributes:
```
Comment (Johaney-s) [inline perun-openapi/openapi.yml:9104]:
```suggestion
  /json/membersManager/getRichMembersNoUserAttributes:
```
This has to match the method name in MembersManager.

File: perun-openapi/openapi.yml [hunk 2]
```diff
@@ -9101,6 +9101,22 @@ paths:
...
+    get:
+      tags:
+        - MembersManager
+      operationId: getRichMembersWithoutUserAttributes
```
Comment (Johaney-s) [inline perun-openapi/openapi.yml:9108]:
```suggestion
      operationId: getRichMembersNoUserAttributes
```

File: perun-core/src/main/java/cz/metacentrum/perun/core/entry/MembersManagerEntry.java [hunk 3]
```diff
@@ -846,6 +846,20 @@ public RichMember getRichMemberWithAttributes(PerunSession sess, Member member)
...
+		Utils.checkPerunSession(sess);
+
+		getPerunBl().getVosManagerBl().checkVoExists(sess, vo);
+
+		// Authorization
+		if (!AuthzResolver.authorizedInternal(sess, "getRichMembers_Vo_policy", vo)) {
+			throw new PrivilegeException(sess, "getRichMembersNoUserAttributes");
+		}
+
+		return getPerunBl().getMembersManagerBl().filterOnlyAllowedAttributes(sess, getMembersManagerBl().getRichMembers(sess, vo), null, true);
```
Comment (Johaney-s) [inline perun-core/src/main/java/cz/metacentrum/perun/core/entry/MembersManagerEntry.java:860]:
```suggestion
		return getPerunBl().getMembersManagerBl().filterOnlyAllowedAttributes(sess, getMembersManagerBl().getRichMembersNoUserAttributes(sess, vo), null, true);
```
we need to call the right method

**Review by Johaney-s — CHANGES_REQUESTED** [review]
it would be nice to have a test for this.

File: perun-core/src/main/java/cz/metacentrum/perun/core/entry/MembersManagerEntry.java [hunk 4]
```diff
@@ -846,6 +846,20 @@ public RichMember getRichMemberWithAttributes(PerunSession sess, Member member)
...
 	}
 
+	@Override
+	public List<RichMember> getRichMembersNoUserAttributes(PerunSession sess, Vo vo) throws PrivilegeException, VoNotExistsException {
+		Utils.checkPerunSession(sess);
+
+		getPerunBl().getVosManagerBl().checkVoExists(sess, vo);
+
+		// Authorization
+		if (!AuthzResolver.authorizedInternal(sess, "getRichMembers_Vo_policy", vo)) {
```
Comment (HejdaJakub) [inline perun-core/src/main/java/cz/metacentrum/perun/core/entry/MembersManagerEntry.java:856]: You should also add the new method to the `perun-roles.yml`. The yml definition will be the same as `getRichMembers_Vo_policy`.
```suggestion
		if (!AuthzResolver.authorizedInternal(sess, "getRichMembersNoUserAttributes_Vo_policy", vo)) {
```

**Review by HejdaJakub — CHANGES_REQUESTED** [review]

**Review by Johaney-s — APPROVED** [review]

**Review by HejdaJakub — APPROVED** [review]

**Pull request merged**